### DISCUSSION TITLE
Welcome to FabCODext Discussions!

### DISCUSSION THREAD

**Discussion by JD-Howard** [AgileBIM/FabCOD#1] [General]
I'll edit this later

## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by DyoungMCWI** [comment]
FYI...Having worked through some of the TypeScript examples for this, that process has given me a better understanding of the underlying object structures. The end result, over a decade ago I stashed away a list of what I thought were undocumented functions that I could never get to work.  Except now I understand how to use them. 

There's going to be a couple dozen...maybe a little less, possibly a up to 3 dozen. Will get them into the TS doc once I test and confirm their use.

**Reply by JD-Howard** [reply]
I have taken this weekend off from FabCOD given I've spent all my free time in the last month going full throttle on it. 

Get those secret functions recorded and create [issues](https://github.com/AgileBIM/FabCOD/issues) for things that aren't working quite right. I'll jump back on feature creation next weekend.

**Reply by DyoungMCWI** [reply]
Have about 1/2 the 30 or so functions added to the TS. Testing the functions is pretty tedious...a couple are very buggy and sporadic at best in trying to determine the proper syntax to call them. Will see if Autodesk has a better explanation. 

Should be complete in a day or two if I can stop myself from looking over prior work and editing my typo's in the comments/documentation.

**Reply by DyoungMCWI** [reply]
Fab Def's TS is done. Added 29 methods. Few minor edits in other notes/comments.

I've pushed it to the Origin in my branch but not made a PR yet. Will try to actually walk through rebuilding the JSON and entire extension process first in the next day or two. If for some reason you want the changes before I get to that, let me know and I'll generate a PR without all that added testing.

PS: Do have a support ticket in w/Autodesk to see if they can clarify a few things that i either don't have just right or are a little buggy. A few more notes in my "undocumented research" doc I've thrown their way and hopefully they'll come back with a few more for me to add.

**Comment by StDistraxion** [comment]
Hello...

This question has probably been asked a thousand times over, but why does an VS (not Code) extension exist just yet?

It looks remarkably close to VB, and besides VS Code is the shell at the heart of VS anyway, right?

So what gives? Just curious, thanks.

**Reply by JD-Howard** [reply]
I don't fully understand your question, but yes COD is very similar to VB because it was clearly based on it. However, MAP Software (the original creators) deviated by only doing a partial implementation and adding a bunch of their own context/behavioral twists. Similar, but not the same. 

RE: VS not VSCode, I've looked into making VS extensions in the past, but it was pretty disgusting. VSCode is a lot more light weight and in one way or another, can or at least has the potential, to do everything VS can do and more. I am sure there are exceptions, but I wouldn't be surprised if the monumental development efforts on VSC don't eventually replace VS. My only reservation about that statement is VSC being a completely open sandbox, which makes injecting VS-Like highly specific context difficult in its current state.

**Comment by StDistraxion** [comment]
I see, well some day :)

But thanks for explaining it a bit further!



On Fri, Mar 26, 2021 at 1:59 PM Joshua Howard ***@***.***>
wrote:

> I don't fully understand your question, but yes COD is very similar to VB
> because it was clearly based on it. However, MAP Software (the original
> creators) deviated by only doing a partial implementation and adding a
> bunch of their own context/behavioral twists. Similar, but not the same.
>
> RE: VS not VSCode, I've looked into making VS extensions in the past, but
> it was pretty disgusting. VSCode is a lot more light weight and in one way
> or another, can or at least has the potential, to do everything VS can do
> and more. I am sure there are exceptions, but I wouldn't be surprised if
> the monumental development efforts on VSC don't eventually VS. My only
> reservation about that statement is VSC being a completely open sandbox,
> which makes injecting VS-Like highly specific context difficult in its
> current state.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/AgileBIM/FabCOD/discussions/1#discussioncomment-535674>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABZNDP7SSDEGGDMG3CN7H73TFTKRRANCNFSM4VKBICRA>
> .
>


-- 
Matthew Anderson
Distraxion Consulting
***@***.*** *email*
(281) 432-7325 *phone*
